plugin/pkg/global/resourcereferencemanager/admission.go

```diff
@@ -585,6 +588,9 @@ func (r *ReferenceManager) Admit(ctx context.Context, a admission.Attributes, _
 								channel <- fmt.Errorf("unable to delete Machine image version '%s/%s' from CloudProfile %q - version is still in use by shoot '%s/%s' by worker %q", worker.Machine.Image.Name, *worker.Machine.Image.Version, cloudProfile.Name, shoot.Namespace, shoot.Name, worker.Name)
 							}
 						}
+						if hasDecreasedLimits {
+							validateShootWorkerLimits(channel, shoot, cloudProfile.Spec.Limits)
```

Comment:
Shouldn't this validation be executed whenever the limit was changed? This would cover the case that a CloudProfile with a low value was created, and a then increased value is validated appropriately.

In addition, `hasDecreasedLimits` is a bit contradicting to the core validation where decreasing directly is forbidden (only through add).

Reply:
I agree that the variable name is misleading. Would you agree to `wasLimitAdded`?
The validation only needs to be executed if a limit has been introduced, not if it is increased or removed again.

Reply:
PTAL at https://github.com/gardener/gardener/pull/11647/commits/5f06be0c046674f1ea100a454c4daa503f5d69dd.